src/Database/TigerBeetle/Raw/Queue.hs

```diff
@@ -1,56 +1,96 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE LambdaCase #-}
 module Database.TigerBeetle.Raw.Queue where
 
-import Control.Concurrent.STM (atomically)
-import Control.Concurrent.STM.TQueue (TQueue, writeTQueue)
+import Database.TigerBeetle.Internal.FFI.Client
+import Database.TigerBeetle.Internal.FFI.Client.ClusterId (ClusterId)
 import Data.Text (Text)
-import Data.Text.Foreign qualified as T
-import Data.Word (Word8)
 import Foreign.Marshal.Alloc (alloca)
-import Foreign.Marshal.Array (peekArray, withArray)
-import Foreign.Storable
-import Database.TigerBeetle.Internal.FFI
+import qualified Data.Text.Foreign as T
+import Foreign (sizeOf, Storable (poke))
+import Control.Exception (finally)
+import Foreign.Ptr (Ptr)
 
--- | Response type for handling callbacks
-data Response = Response
-  { responsePacket :: TBPacket,
-    responseData :: [Word8],
-    responseStatus :: PacketStatus
-  }
+data WithClientOps =
+  WithClientOps
+    { onInitFailure :: TBInitStatus -> IO ()
+    , onDeinit :: TBClientStatus -> IO ()
```

Comment:
These may not be necessary, but I figured its nice to give hooks into the different client lifecycle states.